libsql-server/src/replication/primary/logger.rs

```diff
@@ -84,6 +85,21 @@ unsafe impl WalHook for ReplicationLoggerHook {
         tracing::trace!("Last valid frame before applying: {last_valid_frame}");
         let ctx = Self::wal_extract_ctx(wal);
 
+        let mut binding = ctx.bottomless_replicator.clone();
+        let replicator_mutex: Option<&mut Replicator>;
+        let mut replicator_lock: MutexGuard<Option<Replicator>>;
+        if let Some(replicator) = binding.as_mut() {
+            replicator_lock = replicator.lock().unwrap();
+            if replicator_lock.is_none() {
+                tracing::error!("fatal error: expected a replicator, exiting");
+                std::process::abort()
+            } else {
+                replicator_mutex = Option::from(replicator_lock.as_mut().unwrap());
+            }
+        } else {
+            replicator_mutex = None;
+        }
```

Comment:
in `on_frame`, I wanted to check if bottomless is enabled and if so, hold the lock till the operation is done. Let me know if there is a cleaner way to do above